src/common/image.c

```diff
@@ -804,6 +804,12 @@ void dt_image_update_final_size(const int32_t imgid)
   {
     imgtmp->final_width = ww;
     imgtmp->final_height = hh;
+    if (ww != 0 && hh != 0)
+    {
+      imgtmp->crop_right = ww;
```

Comment:
maybe `width - crop_x` would makes more sense here. To be checked that crop_x is properly defined when calling this routine though.

Reply:
From my understanding we run into the wrong dimensions if `dt_dev_pixelpipe_get_dimensions()` doesn't correctly set `ww` and `hh`. Maybe we have to look there?